scripts/pi-hole/js/utils.js

```diff
@@ -229,13 +229,20 @@ function setBsSelectDefaults() {
   };
 }
 
+var backupStorage = {};
 function stateSaveCallback(itemName, data) {
-  localStorage.setItem(itemName, JSON.stringify(data));
+  if (localStorage === null) {
+    backupStorage[itemName] = JSON.stringify(data);
+  } else {
+    localStorage.setItem(itemName, JSON.stringify(data));
+  }
 }
 
 function stateLoadCallback(itemName) {
   // Receive previous state from client's local storage area
-  var data = localStorage.getItem(itemName);
+  var data =
+    localStorage === null ? backupStorage[itemName] || null : localStorage.getItem(itemName);
```

Comment:
I don't read all your code, but this part
```js
backupStorage[itemName] || null
```
will convert `0` (zero) and `''` (empty string) to `null`.

Is this the intended behavior?

Reply:
Great comment. I pushed a commit to be explicit for undefined to not convert those to null.